src/gui/advancedsettings.h

```diff
@@ -34,13 +34,15 @@
 #include <QSpinBox>
 #include <QTableWidget>
 
-class AdvancedSettings final : public QTableWidget
+#include "guiapplicationcomponent.h"
+
+class AdvancedSettings final : public QTableWidget, public GUIApplicationComponent
```

Comment:
I think it is a bit strange for a dialog/widget/any class to inherit `GUIApplicationComponent`.
The app component is completely unrelated to the class, yet the methods are mixed-in now.

Maybe consider make it a class member/variable instead?

Reply:
> I think it is a bit strange for a dialog/widget/any class to inherit `GUIApplicationComponent`.

I'm sorry, I don't understand, is it strange to you that some classes inherit some base class? `ApplicationComponent` (`GUIApplicationComponent`) is intended to be a base class for all more or less independent "components" of the application in order to ensure their convenient interaction without these "ugly" singletons.
Maybe you're misinterpreting the meaning of the term "component" that I put into it? I mean any more or less independent class (widget, dialog) that is not just a "worker", completely controlled by its owner and does not need any direct relationship with other components.

Reply:
>I'm sorry, I don't understand, is it strange to you that some classes inherit some base class?

IMO not every method from base class is suitable/make sense in the derived class.
`AdvancedSettings` is deriving from QTableWidget so we can see it probably is a widget. However you added another `GUIApplicationComponent`, so what is it now? It is a widget with special power that can access the application directly!? what is this kind of thing? It feels like a Chimera (mythology) that has different heads... which is bad because now there is no clear line on how we expect the derived class would behave.

https://stackoverflow.com/questions/49002/prefer-composition-over-inheritance

I'm under the impression that inheritance should only be done when there is a sense of hierarchy of the functionality so I'm not sure this is way of doing inheritance is widely accepted but it really looks strange to me. Or you do have some strong arguments of needing the inheritance?

Reply:
>However you added another `GUIApplicationComponent`, so what is it now? 

`AdvancedSettings` is `QTableWidget` and `GUIApplicationComponent` at the same time, exactly! 
Just like, for example, I am both a Programmer (this is, of course, too loud to say) and a Parent (of my children). But I'm not a Chimera. I just have several professions/roles (i.e. I inherit several types/interfaces in programming). When my child's school organizes a parent-teacher meeting, it is expected that Parents will come there. So I can come to this meeting because I am a Parent, and maybe my wife, because she is also a Parent (but she is not a Programmer). And Yes, I do not contain/have a Parent, I am a Parent. So aggregation isn't correct choice here (unless you're fanatically denying multiple inheritance).

Reply:
OK, I don't insist too much.